src/main/groovy/edu/oregonstate/mist/locations/frontend/mapper/LocationMapper.groovy

```diff
@@ -0,0 +1,114 @@
+package edu.oregonstate.mist.locations.frontend.mapper
+
+import com.fasterxml.jackson.databind.JsonNode
+import edu.oregonstate.mist.locations.frontend.core.Attributes
+import edu.oregonstate.mist.locations.frontend.core.DayOpenHours
+import edu.oregonstate.mist.locations.frontend.jsonapi.ResourceObject
+import org.joda.time.DateTime
+
+class LocationMapper {
+
+    public ResourceObject map(JsonNode hit) {
+        def source = hit.get("_source")
+
+        def resourceObject = new ResourceObject(
+                id: source.get("id").asText(null),
+                type: source.get("type").asText(null),
+                attributes: getAttributes(source.get("attributes"), hit.get('sort')),
+                links: source.get("links")
+        )
+
+        resourceObject
+    }
+
+    /**
+     * Get attributes from JsonNode
+     * @param attr
+     * @param sort
+     * @return
+     */
+    private static Attributes getAttributes(JsonNode attr, JsonNode sort) {
+
+        Attributes attributes = new Attributes(
+                name: attr.get("name").asText(null),
```

Comment:
I think having a helper private method that you could pass in the field name and returns the output of attr.get($field).asText(null) would make this a bit DRYer.